src/components/components/footer/footer.js

```diff
@@ -1,159 +1,171 @@
 /*
  Contains the footer section.
  */
+import './footer.scss';
 
-// External modules.
 import React from 'react';
+import { connect } from 'react-redux';
+import { mapUserToProps } from './../../../utils/redux-utils';
 import { Link } from 'gatsby';
 import MediaQuery from 'react-responsive';
+import LayoutContained from '../../layouts/layout-contained/layout-contained';
+import Navigation from '../../navigation/footer-navigation/footer-navigation';
 
-// Local modules.
-import LayoutContained from './../../layouts/layout-contained/layout-contained';
-import Navigation from './../../navigation/footer-navigation/footer-navigation';
+class Footer extends React.PureComponent {
```

Comment:
No need to turn this to a full component for a simple !!this.props.user.user, that is used once. It isn't worth the overhead

Reply:
@alexdor I fixed all the other remarks. What do you propose to resolve this one?

Reply:
Turn it back to a function component:
sould be smth like:
```
const Footer = () => {
 // render goes here
}

export default memo(Footer)
```

Reply:
But then we loose the login/logout functionality

Reply:
No need for the functionality to be changed. Have a look at iropal/OSSN#28